src/indra_cogex/client/queries.py

```diff
@@ -928,6 +928,7 @@ def get_evidences_for_stmt_hashes(
     client: Neo4jClient,
     limit: Optional[str] = None,
     remove_medscan: bool = True,
+    mesh_terms: Optional[List[str]] = None,
```

Comment:
Should a similar filter be set up for `get_evidences_for_stmt_hash` as well? I'm not sure if that's really needed, but something to discuss before merging.